src/AutoRest.CSharp/Common/Output/Models/Types/SignatureType.cs

```diff
@@ -0,0 +1,318 @@
+                        break;
+                    }
+                    else
+                    {
+                        parameters.Add(methodParameter);
+                    }
+                }
+
+                // Since we don't have the ability to create the missing types, if any parameter type is missing we can't continue to generate overload methods
+                if (isParameterTypeMissing)
+                {
+                    continue;
+                }
+                result.Add(new MethodSignature(method.Name, null, $"{description}", MapModifiers(method), returnType, null, parameters, IsRawSummayText: true));
+            }
+            return result;
+        }
+
+        private Parameter? FromParameterSymbol(IParameterSymbol parameterSymbol)
+        {
+            var parameterName = parameterSymbol.Name;
+            if (_typeFactory.TryCreateType(parameterSymbol.Type, out var parameterType))
+            {
+                return new Parameter(parameterName, null, parameterType, null, ValidationType.None, null);
+            }
+
+            // If the parameter type can't be found from type factory, the type was removed from current version
+            // Since we don't have the ability to create the missing types, we can't continue to generate overload methods
+            return null;
+        }
+
+        private static MethodSignatureModifiers MapModifiers(IMethodSymbol methodSymbol)
+        {
+            var modifiers = MethodSignatureModifiers.None;
+            var accessibility = methodSymbol.DeclaredAccessibility;
+            switch (accessibility)
+            {
+                case Accessibility.Public:
+                    modifiers |= MethodSignatureModifiers.Public;
+                    break;
+                case Accessibility.Internal:
+                    modifiers |= MethodSignatureModifiers.Internal;
+                    break;
+                case Accessibility.Private:
+                    modifiers |= MethodSignatureModifiers.Private;
+                    break;
+                case Accessibility.Protected:
+                    modifiers |= MethodSignatureModifiers.Protected;
+                    break;
+                case Accessibility.ProtectedAndInternal:
+                    modifiers |= MethodSignatureModifiers.Protected | MethodSignatureModifiers.Internal;
+                    break;
+            }
+            if (methodSymbol.IsStatic)
+            {
+                modifiers |= MethodSignatureModifiers.Static;
+            }
+            if (methodSymbol.IsAsync)
+            {
+                modifiers |= MethodSignatureModifiers.Async;
+            }
+            if (methodSymbol.IsVirtual)
+            {
+                modifiers |= MethodSignatureModifiers.Virtual;
+            }
+            if (methodSymbol.IsOverride)
+            {
+                modifiers |= MethodSignatureModifiers.Override;
+            }
+            return modifiers;
+        }
+    }
```

Comment:
we have something similar here: https://github.com/Azure/autorest.csharp/blob/6f0917d17343ee5f12d60636f14e74d26783f44e/src/AutoRest.CSharp/Common/Output/Models/Types/ModelTypeProviderFields.cs#L311
maybe we could find somewhere like `BuilderHelpers` to have a generic method for this purpose and just use it here.

Reply:
The existing one only get accessibility, what about the others?

Reply:
Moved this to `BuilderHelpers`, the only usage is here for now.